plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/StreamDataImporterColumnInfo.java

```diff
@@ -39,9 +42,26 @@ public StreamDataImporterColumnInfo(StreamEntityMapping entity, int columnIndex,
         this.dataKind = dataKind;
     }
 
-    public void updateMaxLength(long maxLength) {
-        if (getMaxLength() < maxLength) {
-            setMaxLength(roundToNextPowerOf2(maxLength));
+    public void updateMaxLength(@Nullable DBPDataSource dataSource, long maxLengthFromData) {
+        long maxLength = getMaxLength();
+        DBPPreferenceStore globalPreferenceStore = DTActivator.getDefault().getPreferences();
+        if (dataSource != null) {
+            // First check data source settings for max data type length
+            DBPPreferenceStore dataSourcePreferenceStore = dataSource.getContainer().getPreferenceStore();
+            if (dataSourcePreferenceStore.contains(DTConstants.PREF_MAX_TYPE_LENGTH) &&
```

Comment:
Move  dataSourcePreferenceStore.getInt(DTConstants.PREF_MAX_TYPE_LENGTH) to a variable